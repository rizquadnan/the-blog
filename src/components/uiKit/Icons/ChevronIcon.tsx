import { Icon, IconProps } from '@chakra-ui/icons'
import React from 'react'

export function ChevronIcon(props: IconProps) {
  return (
    <Icon
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 24 24"
      strokeWidth={1.5}
      stroke="currentColor"
      {...props}
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M15.75 19.5L8.25 12l7.5-7.5"
      />
    </Icon>
  )
}

export default ChevronIcon
