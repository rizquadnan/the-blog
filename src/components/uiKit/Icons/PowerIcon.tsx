import { Icon, IconProps } from '@chakra-ui/icons'
import React from 'react'

export function PowerIcon(props: IconProps) {
  return (
    <Icon
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 24 24"
      strokeWidth={1.5}
      stroke="currentColor"
      {...props}
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M5.636 5.636a9 9 0 1012.728 0M12 3v9"
      />
    </Icon>
  )
}

export default PowerIcon
