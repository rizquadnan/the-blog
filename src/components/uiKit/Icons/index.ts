export * from "./SunIcon";
export * from "./MoonIcon";
export * from "./HomeIcon";
export * from "./UserGroupIcon";
export * from "./SettingsIcon";
export * from "./PowerIcon";
export * from "./EditIcon";
