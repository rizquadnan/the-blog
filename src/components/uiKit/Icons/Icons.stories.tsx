import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { Icons } from './Icons'

type TIcons = React.ComponentProps<typeof Icons>
export default {
  title: 'DataDisplay/Icons',
  component: Icons,
} as Meta<TIcons>

const Template: Story<TIcons> = () => <Icons />

export const AllIcons = Template.bind({})
