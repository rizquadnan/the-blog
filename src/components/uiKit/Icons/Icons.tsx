import {
  Wrap,
  WrapItem as _WrapItem,
  WrapItemProps as _WrapItemProps,
} from '@chakra-ui/react'
import React from 'react'
import ChevronIcon from './ChevronIcon'
import EditIcon from './EditIcon'
import HomeIcon from './HomeIcon'
import MoonIcon from './MoonIcon'
import PowerIcon from './PowerIcon'
import SendIcon from './SendIcon'
import SettingsIcon from './SettingsIcon'
import SunIcon from './SunIcon'
import UserGroupIcon from './UserGroupIcon'

function WrapItem(props: _WrapItemProps) {
  return (
    <_WrapItem
      border="1px solid"
      borderColor="gray.300"
      padding="12px"
      borderRadius="4px"
      {...props}
    />
  )
}

export function Icons() {
  return (
    <Wrap spacing="16px">
      <WrapItem>
        <SunIcon />
      </WrapItem>
      <WrapItem>
        <MoonIcon />
      </WrapItem>
      <WrapItem>
        <HomeIcon />
      </WrapItem>
      <WrapItem>
        <UserGroupIcon />
      </WrapItem>
      <WrapItem>
        <SettingsIcon />
      </WrapItem>
      <WrapItem>
        <PowerIcon />
      </WrapItem>
      <WrapItem>
        <EditIcon />
      </WrapItem>
      <WrapItem>
        <SendIcon />
      </WrapItem>
      <WrapItem>
        <ChevronIcon />
      </WrapItem>
    </Wrap>
  )
}

export default Icons
