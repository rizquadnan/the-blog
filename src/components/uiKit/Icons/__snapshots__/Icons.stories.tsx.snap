// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Display/Icons AllIcons smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": undefined,
  "name": "data-display-icons--all-icons",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
